sources/api/pluto/src/main.rs

```diff
@@ -365,16 +370,25 @@ async fn generate_private_dns_name(
         .context(error::ImdsNoneSnafu {
             what: "instance ID",
         })?;
-    aws_k8s_info.hostname_override = Some(
-        ec2::get_private_dns_name(
-            region,
-            &instance_id,
-            aws_k8s_info.https_proxy.clone(),
-            aws_k8s_info.no_proxy.clone(),
-        )
-        .await
-        .context(error::Ec2Snafu)?,
-    );
+
+    match hostname_source {
+        KubernetesHostnameOverrideSource::PrivateDNSName => {
+            aws_k8s_info.hostname_override = Some(
+                ec2::get_private_dns_name(
+                    region,
+                    &instance_id,
+                    aws_k8s_info.https_proxy.clone(),
+                    aws_k8s_info.no_proxy.clone(),
+                )
+                .await
+                .context(error::Ec2Snafu)?,
+            );
+        }
+        KubernetesHostnameOverrideSource::InstanceID => {
+            aws_k8s_info.hostname_override = Some(instance_id);
+        }
```

Comment:
nit: it'd be slightly better to fetch the instance ID within this block, as otherwise we pay for the cost of an IMDS interaction and end up throwing away the result when using the private DNS source

Reply:
We use it both places, in the private DNS lookup we filter by the instance ID so we only get the single response.